src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Media/MediaPlayerState.cs

```diff
@@ -877,7 +877,7 @@ private void OpenMedia(Uri source)
                 {
                     // target is HTTPS. Then, elevate ONLY if we are NOT coming from HTTPS (=XDomain HTTPS app to HTTPS media disallowed)
                     Uri appDeploymentUri = SecurityHelper.ExtractUriForClickOnceDeployedApp();
-                    if (!SecurityHelper.AreStringTypesEqual(appDeploymentUri.Scheme, Uri.UriSchemeHttps))
+                    if (appDeploymentUri != null && !SecurityHelper.AreStringTypesEqual(appDeploymentUri.Scheme, Uri.UriSchemeHttps))
```

Comment:
For the change as proposed, why not simply this? It just needs to fall-through into evaluating `elevated=true` one way or another. 

```C#
!SecurityHelper.AreStringTypesEqual(appDeploymentUri?.Scheme, Uri?.UriSchemeHttps)
```

Reply:
I find `Uri?.UriSchemeHttps` odd, as it would never be null. I like the above better as it clearly reads "only check this if there's an appDeploymentUri"